### PR TITLE
Implement User login authentication

### PR DESCRIPTION
**Pull request by Ndifreke**
#### What does this PR do?
Implement User login authentication

#### Description of Task to be completed?
* Generate JSON web token for a successful login

#### What are the relevant pivotal tracker stories?
[#163426581](https://www.pivotaltracker.com/n/projects/2240406/stories/163426581)

### CODE REVIEW

File: server/controllers/userController.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ * @description Generate login access token for a user
+ *
+ * @param {Request}  req
+ * @param {httpResponse} res
+ * @param {function} next
+ * @returns {object} HttpResponse {statusCode:int, success:boolean, data:object, message:string}
+ */
+export const userLogin = async (req, res, next) => {
+  const { email, password } = req.body;
+  if (email && password) {
```
Comment (eokwukwe) [inline server/controllers/userController.js]: Add `try catch` in the login controller to handle any other possible error.

File: server/controllers/userController.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import database from "../models/index";
+
+import controller from "../helpers/index";
+
+const { comparePassword, generateJwtToken, httpResponse } = controller;
```
Comment (mrkingz) [inline server/controllers/userController.js]: What is generateJwtToken, why not generateJWT, as the former is somehow repetitive

File: server/controllers/userController.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      }
+    }
+  }
+  return httpResponse(res, {
+    statusCode: 401,
+    success: false,
+    message: "Invalid login credential",
+    data: null
+  });
+};
```
Comment (mrkingz) [inline server/controllers/userController.js]: No try-catch, why?

File: server/helpers/jwt.js [hunk 4]
```diff
@@ -3,19 +3,14 @@ import jwt from "jsonwebtoken";
...
-const generateJwtToken = user => {
-  const { id, email, isAdmin } = user;
-  return jwt.sign(
-    {
-      id,
-      email,
-      isAdmin
-    },
-    process.env.JWT_SECRET
-  );
```
Comment (mrkingz) [inline server/helpers/jwt.js]: Please, consider signing jwt with an issuer and subject.

**Review by eokwukwe — APPROVED** [review]

**Review by obasajujoshua31 — APPROVED** [review]

**Review by mrkingz — APPROVED** [review]

**Review by ODINAKACHUKWU — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Comment by benfluleck** [comment]
Why is my commit in your workflow? Also implement @eokwukwe validation on your branch

**Pull request closed without merging**